consul/leader.go

```diff
@@ -370,12 +374,19 @@ func (s *Server) handleAliveMember(member serf.Member) error {
 	}
 
 	// Check if the node exists
-	_, found, addr := state.GetNode(member.Name)
-	if found && addr == member.Addr.String() {
+	state := s.fsm.State()
+	_, node, err := state.GetNode(member.Name)
+	if err != nil {
+		return fmt.Errorf("failed to lookup node %s: %s", member.Name, err)
```

Comment:
It seems weird that this is the only error we prefix. Ideally they would all have some context around them.